Cotecna.OpenSource.Domain.Core/Common/ApplicationMediator.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+
+namespace Cotecna.OpenSource.Domain.Core
+{
+    public interface ApplicationMediator
+    {
+        /// <summary>
+        /// Defines a mediator to encapsulate command and querys.
+        /// Represents <see cref="IApplicationMediator"/> Injectable Service Interface
+        /// </summary>
+        public interface IApplicationMediator
+        {
+            bool Dispatch(ICommand command);
+
+            T Dispatch<T>(IQuery<T> query);
+        }
+
+        /// <summary>
+        /// Defines a instance of IApplicationMediator <see cref="IApplicationMediator"/> 
+        /// </summary>
+        public sealed class ApplicationMediator : IApplicationMediator
+        {
+            private readonly IServiceProvider _provider;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="IApplicationMediator"/> class.
+            /// </summary>
+            /// <param name="provider">The single instance of <see cref="IServiceProvider"/></param>
+            public ApplicationMediator(IServiceProvider provider)
+            {
+                _provider = provider;
+            }
+
+            /// <summary>
```

Comment:
perhaps is better to move this documentation, to the interface since normally, implementation will not be directly referenced

Reply:
Moved!